Demo/src/main/java/com/paypal/android/ui/paypalbuttons/PayPalButtonsView.kt

```diff
@@ -78,39 +84,74 @@ fun PayPalButtonsView(viewModel: PayPalButtonsViewModel = viewModel()) {
                 )
             }
-            Spacer(modifier = Modifier.size(8.dp))
             PaymentButtonShapeOptionList(
                 selectedOption = uiState.paymentButtonShape,
                 onSelection = { value -> viewModel.paymentButtonShape = value }
             )
-            Spacer(modifier = Modifier.size(4.dp))
-            Text(
-                text = "Custom Corner Radius",
-                color = Color.Black,
-                style = MaterialTheme.typography.titleLarge,
-            )
-            Spacer(modifier = Modifier.size(8.dp))
-            Slider(
-                value = uiState.customCornerRadius ?: 0.0f,
-                valueRange = 0f..CORNER_RADIUS_SLIDER_MAX,
+            CustomCornerRadiusSlider(
+                value = uiState.customCornerRadius,
                 onValueChange = { value -> viewModel.customCornerRadius = value }
             )
-            Spacer(modifier = Modifier.size(8.dp))
             PaymentButtonSizeOptionList(
                 selectedOption = uiState.paymentButtonSize,
                 onSelection = { value -> viewModel.paymentButtonSize = value }
             )
+            Spacer(modifier = Modifier.size(contentPadding))
         }
     }
 }
 
-const val CORNER_RADIUS_SLIDER_MAX = 100.0f
+@Composable
+fun CustomCornerRadiusSlider(value: Float?, onValueChange: (Float) -> Unit) {
+    Card {
+        Row(
+            modifier = Modifier
+                .background(MaterialTheme.colorScheme.inverseSurface)
+        ) {
+            Text(
+                text = "Custom Corner Radius",
+                color = MaterialTheme.colorScheme.inverseOnSurface,
+                style = MaterialTheme.typography.bodyLarge,
+                modifier = Modifier
+                    .padding(UIConstants.paddingMedium)
+                    .fillMaxWidth()
+            )
+        }
+        Column(
+            modifier = Modifier.padding(
+                horizontal = UIConstants.paddingLarge,
+                vertical = UIConstants.paddingMedium
+            )
+        ) {
+            Text(
+                text = value?.let { "${value}px" } ?: "UNSET",
+                color = MaterialTheme.colorScheme.onSurfaceVariant,
+                style = MaterialTheme.typography.titleLarge,
+                modifier = Modifier
+                    .padding(bottom = UIConstants.paddingSmall)
+                    .fillMaxWidth()
+            )
+            Slider(
+                value = value ?: 0.0f,
+                valueRange = 0f..CORNER_RADIUS_SLIDER_MAX,
+                steps = CORNER_RADIUS_SLIDER_MAX.toInt(),
+                colors = SliderDefaults.colors(
+                    inactiveTrackColor = MaterialTheme.colorScheme.onSurfaceVariant,
+                ),
+                onValueChange = { value ->
+                    // round to integer
+                    val roundedValue = value.roundToInt().toFloat()
```

Comment:
Is there a reason why we need to round the value to an integer? Would it be better to have this rounding logic live in the ViewModel where `viewModel.customCornerRadius` is set?

Reply:
:nod: good call yeah, that's a much better place to put the rounding logic.

Reply:
After some refactoring, ended up landing on an `IntSlider` to encapsulate the conversion logic (in [e3374da](https://github.com/paypal/paypal-android/pull/222/commits/e3374da17a5595f9a6b22288c5b30131d1740ec7)). Now the view model will only see `Int` types.